hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/math/AbsFunctionIntegrationTest.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.math;
+
+import com.hazelcast.sql.SqlColumnType;
+import com.hazelcast.sql.SqlErrorCode;
+import com.hazelcast.sql.impl.expression.SqlExpressionIntegrationTestSupport;
+import com.hazelcast.sql.support.expressions.ExpressionValue;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class AbsFunctionIntegrationTest extends SqlExpressionIntegrationTestSupport {
```

Comment:
Looks like there are no tests for boolean columns and parameters here and for other expressions as well.

Reply:
Let's collect the list of missing test cases and add them separately. Subsequent PRs fixes more problems with conversions, so it would be waste of time adding these tests in this PR.

Reply:
Ticket: https://github.com/hazelcast/hazelcast/issues/17360

Reply:
Not sure how it does relate to problematic conversions, boolean-to-numeric conversion is prohibited currently and there are no plans I'm aware of to change that, it's just the tests missing, that should do be easy to add them.

Reply:
In this specific branch parameter conversions do not work properly - it allows conversions that are otherwise explicit only. Subsequent PRs fixes that.

Reply:
Moreover, we have broken type inference - it also allows conversions that are otherwise explicit (e.g. INT to VARCHAR). We do not have a solution for this at the moment, neither we know the exact error messages that we will have in the future. 
That said, these corner cases should not be critical for beta IMO

Reply:
Does that mean boolean can be converted to numeric for `ABS` in this PR?

Reply:
No, it can't. But the error message that appears in this branch is different, from what we will have after the next PR is ready.

Reply:
In which next PR the message would change? It shouldn't be hard to add the tests in this PR and fix a couple of error message strings here and there later. The actual error messages are not important, the fact that we are failing while converting boolean to numeric is important.

Reply:
In the next PR with predicates. Why should we spend time on writing tests only to rewrite them later? There is no BOOLEAN-specific logic in this PR, all expressions work the same way and rely on the existing expressions code base. 

The more expressions I implement, the more bugs I found. This requires to me rewrite tests and spend precious time that we have before code freeze. We will easily add more tests after the code freeze. 

Please do not follow the formal approach here. Our goal is to implement as many expressions as possible *before* the code freeze with sensible quality. In this specific case, we know for sure that there is no BOOLEAN-specific code, so the risk is minimal; and that the tests we write now will have to be rewritten afterward, so it is definitely a waste of time.